WordPress/src/main/java/org/wordpress/android/ui/domains/DomainRegistrationActivity.kt

```diff
@@ -1,17 +1,33 @@
 package org.wordpress.android.ui.domains
 
+import android.app.Activity
 import android.os.Bundle
 import android.view.MenuItem
 import androidx.appcompat.app.AppCompatActivity
 import kotlinx.android.synthetic.main.toolbar.*
 import org.wordpress.android.R
+import org.wordpress.android.ui.domains.DomainRegistrationActivity.DomainRegistrationPurpose.CTA_DOMAIN_CREDIT_REDEMPTION
 
 class DomainRegistrationActivity : AppCompatActivity(), DomainRegistrationStepsListener {
+    enum class DomainRegistrationPurpose {
+        AUTOMATED_TRANSFER,
+        CTA_DOMAIN_CREDIT_REDEMPTION
+    }
+
+    companion object {
+        const val DOMAIN_REGISTRATION_PURPOSE_KEY = "DOMAIN_REGISTRATION_PURPOSE_KEY"
+    }
+
+    var domainRegistrationPurpose: DomainRegistrationPurpose? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         setContentView(R.layout.activity_domain_suggestions_activity)
 
+        domainRegistrationPurpose = intent.getSerializableExtra(DOMAIN_REGISTRATION_PURPOSE_KEY)
+                as DomainRegistrationPurpose
```

Comment:
Good catch! I added a null guard usin `as?` and left it nullable for now, since it will be called using `viewDomainRegistrationActivity` from `develop` branch, and I want to reduce conflicts when merging this back into it.

I'll change it to `lateinit` in next domain related PR 👍